examples/pxScene2d/src/rcvrcore/optimus.js

```diff
@@ -381,12 +446,21 @@ function Application(props) {
     _externalApp.on("onClientStopped", function () { _this.log( "onClientStopped"); }); // is never called
     _externalApp.ready.then(function() {
       _this.log("successfully created Spark app: " + _this.id);
-      _readyResolve();
+      _readyBaseResolve();
       _this.applicationReady();
+
+      if(typeof(_externalApp.api.uiReady) == "object")
+      {
+        _externalApp.api.uiReady.then(
+          function(result) { _uiReadyResolve(result); }, 
+          function(err) { _uiReadyReject(err); }
+        );
+      }
     }, function rejection() {
       var msg = "Failed to load uri: " + uri;
       _this.log("failed to launch Spark app: " + _this.id + ". " + msg);
-      _readyReject(new Error("failed to create. " + msg));
+      _readyBaseReject(new Error("failed to create. " + msg));
+      _uiReadyReject();
```

Comment:
Do we need to add a check here to confirm _readyBaseReject and _uiReadyReject are valid before calling them?  Can they still be undefined at this point?